src/java/com/google/tagpost/spanner/DataService.java

```diff
@@ -8,9 +8,9 @@
 
   List<Thread> getAllThreadsByTag(String tag);
 
-  Thread addNewThreadWithTag(Thread thread) throws Exception;
+  Thread addNewThreadWithTag(Thread thread) throws DataException;
```

Comment:
RuntimeException is not checked exception, so it does not need to be specified here. Just remove "throw xxx".